[CI][ML] MlDistributedFailureIT.testLoseDedicatedMasterNode randomly fails on feature-jindex-master branch

This test has been observed to fail occasionally in the `feature-jindex-master` branch. I have not yet managed to reproduce locally. I will shortly be muting the test as we need a green build to merge the branch in master. However, I am raising the issue to ensure we get to the bottom of this failure.

Link to failure (one of them): https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/2556/console

Reproduce with:

```
./gradlew :x-pack:plugin:ml:internalClusterTest -Dtests.seed=D2A618A38265651F -Dtests.class=org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT -Dtests.method="testLoseDedicatedMasterNode" -Dtests.security.manager=true -Dtests.locale=be-BY -Dtests.timezone=Asia/Katmandu -Dcompiler.java=11 -Druntime.java=8
```

Failure:

```
> Throwable #1: java.lang.AssertionError
   > 	at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:847)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:821)
   > 	at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.run(MlDistributedFailureIT.java:292)
   > 	at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testLoseDedicatedMasterNode(MlDistributedFailureIT.java:88)
   > 	at java.lang.Thread.run(Thread.java:748)
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 more
   > 	Suppressed: java.lang.AssertionError
   > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
   > 		... 40 moreThrowable #2: java.lang.RuntimeException: Had to resort to force-stopping datafeed, something went wrong?
   > 	at org.elasticsearch.xpack.ml.support.BaseMlIntegTestCase.deleteAllDatafeeds(BaseMlIntegTestCase.java:296)
   > 	at org.elasticsearch.xpack.ml.support.BaseMlIntegTestCase.cleanupWorkaround(BaseMlIntegTestCase.java:209)
   > 	at java.lang.Thread.run(Thread.java:748)
   > Caused by: java.util.concurrent.ExecutionException: ElasticsearchStatusException[Cannot stop datafeed [data_feed_id] because the datafeed does not have an assigned node. Use force stop to stop the datafeed]
   > 	at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:266)
   > 	at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:253)
   > 	at org.elasticsearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:87)
   > 	at org.elasticsearch.xpack.ml.support.BaseMlIntegTestCase.deleteAllDatafeeds(BaseMlIntegTestCase.java:284)
   > 	... 36 more
   > Caused by: ElasticsearchStatusException[Cannot stop datafeed [data_feed_id] because the datafeed does not have an assigned node. Use force stop to stop the datafeed]
   > 	at org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper.conflictStatusException(ExceptionsHelper.java:50)
   > 	at org.elasticsearch.xpack.ml.action.TransportStopDatafeedAction.normalStopDatafeed(TransportStopDatafeedAction.java:147)
   > 	at org.elasticsearch.xpack.ml.action.TransportStopDatafeedAction.lambda$doExecute$0(TransportStopDatafeedAction.java:130)
   > 	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:60)
   > 	at org.elasticsearch.xpack.ml.datafeed.persistence.DatafeedConfigProvider.lambd
```

This just failed on `6.x` as well, with a matching stack trace. Doesn't reproduce.

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/696/consoleFull
```
./gradlew :x-pack:plugin:ml:internalClusterTest -Dtests.seed=CD15640088E44B91 -Dtests.class=org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT -Dtests.method="testLoseDedicatedMasterNode" -Dtests.security.manager=true -Dtests.locale=de-CH -Dtests.timezone=Poland -Dcompiler.java=11 -Druntime.java=8
```

```
01:24:58   1> [2018-12-19T02:23:49,064][INFO ][o.e.x.m.i.MlDistributedFailureIT] [testLoseDedicatedMasterNode] after test
01:24:58 ERROR   24.7s J1 | MlDistributedFailureIT.testLoseDedicatedMasterNode <<< FAILURES!
01:24:58    > Throwable #1: java.lang.AssertionError
01:24:58    > 	at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:848)
01:24:58    > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:822)
01:24:58    > 	at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.run(MlDistributedFailureIT.java:292)
01:24:58    > 	at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testLoseDedicatedMasterNode(MlDistributedFailureIT.java:88)
01:24:58    > 	at java.lang.Thread.run(Thread.java:748)
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 more
01:24:58    > 	Suppressed: java.lang.AssertionError
01:24:58    > 		at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$15(MlDistributedFailureIT.java:298)
01:24:58    > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
01:24:58    > 		... 40 moreThrowable #2: java.lang.RuntimeException: Had to resort to force-stopping datafeed, something went wrong?
01:24:58    > 	at org.elasticsearch.xpack.ml.support.BaseMlIntegTestCase.deleteAllDatafeeds(BaseMlIntegTestCase.java:295)
01:24:58    > 	at org.elasticsearch.xpack.ml.support.BaseMlIntegTestCase.cleanupWorkaround(BaseMlIntegTestCase.java:208)
01:24:58    > 	at java.lang.Thread.run(Thread.java:748)
01:24:58    > Caused by: java.util.concurrent.ExecutionException: ElasticsearchStatusException[Cannot stop datafeed [data_feed_id] because the datafeed does not have an assigned node. Use force stop to stop the datafeed]
01:24:58    > 	at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:265)
01:24:58    > 	at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:252)
01:24:58    > 	at org.elasticsearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:94)
01:24:58    > 	at org.elasticsearch.xpack.ml.support.BaseMlIntegTestCase.deleteAllDatafeeds(BaseMlIntegTestCase.java:283)
01:24:58    > 	... 36 more
01:24:58    > Caused by: ElasticsearchStatusException[Cannot stop datafeed [data_feed_id] because the datafeed does not have an assigned node. Use force stop to stop the datafeed]
01:24:58    > 	at org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper.conflictStatusException(ExceptionsHelper.java:50)
01:24:58    > 	at org.elasticsearch.xpack.ml.action.TransportStopDatafeedAction.normalStopDatafeed(TransportStopDatafeedAction.java:153)
01:24:58    > 	at org.elasticsearch.xpack.ml.action.TransportStopDatafeedAction.lambda$doExecute$0(TransportStopDatafeedAction.java:136)
01:24:58    > 	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:60)
01:24:58    > 	at org.elasticsearch.xpack.ml.datafeed.DatafeedConfigReader.lambda$expandDatafeedIds$1(DatafeedConfigReader.java:85)
01:24:58    > 	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:60)
01:24:58    > 	at org.elasticsearch.xpack.ml.datafeed.persistence.DatafeedConfigProvider.lambda$expandDatafeedIdsWithoutMissingCheck$4(DatafeedConfigProvider.java:422)
01:24:58    > 	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:60)
01:24:58    > 	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)
01:24:58    > 	at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:85)
01:24:58    > 	at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:81)
01:24:58    > 	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onResponse(AbstractSearchAsyncAction.java:313)
01:24:58    > 	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onResponse(AbstractSearchAsyncAction.java:50)
01:24:58    > 	at org.elasticsearch.action.search.FetchSearchPhase$3.run(FetchSearchPhase.java:213)
01:24:58    > 	at org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:160)
01:24:58    > 	at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:153)
01:24:58    > 	at org.elasticsearch.action.search.ExpandSearchPhase.run(ExpandSearchPhase.java:119)
01:24:58    > 	at org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:160)
01:24:58    > 	at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:153)
01:24:58    > 	at org.elasticsearch.action.search.FetchSearchPhase.moveToNextPhase(FetchSearchPhase.java:206)
01:24:58    > 	at org.elasticsearch.action.search.FetchSearchPhase.lambda$innerRun$2(FetchSearchPhase.java:104)
01:24:58    > 	at org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:110)
01:24:58    > 	at org.elasticsearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:44)
01:24:58    > 	at org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:86)
01:24:58    > 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:759)
01:24:58    > 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
01:24:58    > 	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:41)
01:24:58    > 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
01:24:58    > 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
01:24:58    > 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
01:24:58    > 	... 1 more
```

--- a/server/src/main/java/org/elasticsearch/persistent/PersistentTasksClusterService.java
+++ b/server/src/main/java/org/elasticsearch/persistent/PersistentTasksClusterService.java
@@ -51,7 +51,7 @@ public class PersistentTasksClusterService implements ClusterStateListener, Clos
 
     public static final Setting<TimeValue> CLUSTER_TASKS_ALLOCATION_RECHECK_INTERVAL_SETTING =
         Setting.timeSetting("cluster.persistent_tasks.allocation.recheck_interval", TimeValue.timeValueSeconds(30),
-            TimeValue.timeValueSeconds(10), Setting.Property.Dynamic, Setting.Property.NodeScope);
+            TimeValue.timeValueSeconds(1), Setting.Property.Dynamic, Setting.Property.NodeScope);
 
     private static final Logger logger = LogManager.getLogger(PersistentTasksClusterService.class);
 

--- a/server/src/main/java/org/elasticsearch/persistent/PersistentTasksClusterService.java
+++ b/server/src/main/java/org/elasticsearch/persistent/PersistentTasksClusterService.java
@@ -51,7 +51,7 @@ public class PersistentTasksClusterService implements ClusterStateListener, Clos
 
     public static final Setting<TimeValue> CLUSTER_TASKS_ALLOCATION_RECHECK_INTERVAL_SETTING =
         Setting.timeSetting("cluster.persistent_tasks.allocation.recheck_interval", TimeValue.timeValueSeconds(30),
-            TimeValue.timeValueSeconds(1), Setting.Property.Dynamic, Setting.Property.NodeScope);
+            TimeValue.timeValueSeconds(10), Setting.Property.Dynamic, Setting.Property.NodeScope);
 
     private static final Logger logger = LogManager.getLogger(PersistentTasksClusterService.class);
 
@@ -73,7 +73,8 @@ public class PersistentTasksClusterService implements ClusterStateListener, Clos
             this::setRecheckInterval);
     }
 
-    void setRecheckInterval(TimeValue recheckInterval) {
+    // visible for testing only
+    public void setRecheckInterval(TimeValue recheckInterval) {
         periodicRechecker.setInterval(recheckInterval);
     }
 

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
@@ -59,7 +59,6 @@ public class MlDistributedFailureIT extends BaseMlIntegTestCase {
     protected Settings nodeSettings(int nodeOrdinal) {
         return Settings.builder().put(super.nodeSettings(nodeOrdinal))
             .put(MachineLearning.CONCURRENT_JOB_ALLOCATIONS.getKey(), 4)
-            .put(PersistentTasksClusterService.CLUSTER_TASKS_ALLOCATION_RECHECK_INTERVAL_SETTING.getKey(), TimeValue.timeValueSeconds(1))
             .build();
     }
 
@@ -292,6 +291,17 @@ public class MlDistributedFailureIT extends BaseMlIntegTestCase {
         client().admin().indices().prepareSyncedFlush().get();
 
         disrupt.run();
+
+        PersistentTasksClusterService persistentTasksClusterService =
+            internalCluster().getInstance(PersistentTasksClusterService.class, internalCluster().getMasterName());
+        // Speed up rechecks to a rate that is quicker than what settings would allow.
+        // The tests would work eventually without doing this, but the assertBusy() below
+        // would need to wait 30 seconds, which would make the suite run very slowly.
+        // The 1 second refresh puts a greater burden on the master node to recheck
+        // persistent tasks, but it will cope in these tests as it's not doing anything
+        // else.
+        persistentTasksClusterService.setRecheckInterval(TimeValue.timeValueSeconds(1));
+
         assertBusy(() -> {
             ClusterState clusterState = client().admin().cluster().prepareState().get().getState();
             PersistentTasksCustomMetaData tasks = clusterState.metaData().custom(PersistentTasksCustomMetaData.TYPE);

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
@@ -15,12 +15,14 @@ import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.ByteSizeUnit;
 import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.common.xcontent.DeprecationHandler;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
 import org.elasticsearch.common.xcontent.XContentHelper;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.common.xcontent.XContentType;
 import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.persistent.PersistentTasksClusterService;
 import org.elasticsearch.persistent.PersistentTasksCustomMetaData;
 import org.elasticsearch.persistent.PersistentTasksCustomMetaData.PersistentTask;
 import org.elasticsearch.test.junit.annotations.TestLogging;
@@ -57,6 +59,7 @@ public class MlDistributedFailureIT extends BaseMlIntegTestCase {
     protected Settings nodeSettings(int nodeOrdinal) {
         return Settings.builder().put(super.nodeSettings(nodeOrdinal))
             .put(MachineLearning.CONCURRENT_JOB_ALLOCATIONS.getKey(), 4)
+            .put(PersistentTasksClusterService.CLUSTER_TASKS_ALLOCATION_RECHECK_INTERVAL_SETTING.getKey(), TimeValue.timeValueSeconds(1))
             .build();
     }
 
@@ -72,7 +75,6 @@ public class MlDistributedFailureIT extends BaseMlIntegTestCase {
         });
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/32905")
     public void testLoseDedicatedMasterNode() throws Exception {
         internalCluster().ensureAtMostNumDataNodes(0);
         logger.info("Starting dedicated master node...");

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
@@ -297,10 +297,10 @@ public class MlDistributedFailureIT extends BaseMlIntegTestCase {
         // Speed up rechecks to a rate that is quicker than what settings would allow.
         // The tests would work eventually without doing this, but the assertBusy() below
         // would need to wait 30 seconds, which would make the suite run very slowly.
-        // The 1 second refresh puts a greater burden on the master node to recheck
+        // The 200ms refresh puts a greater burden on the master node to recheck
         // persistent tasks, but it will cope in these tests as it's not doing anything
         // else.
-        persistentTasksClusterService.setRecheckInterval(TimeValue.timeValueSeconds(1));
+        persistentTasksClusterService.setRecheckInterval(TimeValue.timeValueMillis(200));
 
         assertBusy(() -> {
             ClusterState clusterState = client().admin().cluster().prepareState().get().getState();

